File: Programmars/Level_1/두 개 뽑아서 더하기.py
def DFS(level, start):
    if level == 2:
        answer.append(sum(result))
    else:
        for i in range(start, len(n)):
            result[level] = n[i]
            DFS(level+1, i+1)


def solution(numbers):
    global n, answer, result
    n = numbers
    answer = []
    result = [0]*3
    DFS(0, 0)  # 조합(DFS)을 구하지 않고 이중 for문으로 구해도 된다
    answer = list(sorted(set(answer)))
    return answer


print(solution([2, 1, 3, 4, 1]))  # [2,3,4,5,6,7]
print(solution([5, 0, 2, 7]))  # [2,5,7,9,12]

File: JavaScript/Programmars/Level_1/두 개 뽑아서 더하기.js
function solution(numbers) {
  const answer = new Set();
  const len = numbers.length;
  for (let i = 0; i < len; i++) {
    for (let j = i + 1; j < len; j++) {
      answer.add(numbers[i] + numbers[j]);
    }
  }

  return [...answer].sort((a, b) => a - b);
}

console.log(solution([2, 1, 3, 4, 1])); // [2,3,4,5,6,7]
console.log(solution([5, 0, 2, 7])); // [2,5,7,9,12]
